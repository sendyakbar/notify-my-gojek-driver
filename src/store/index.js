import { createStore, combineReducers } from 'redux'

// reducers
import List from '../reducers/list'
import Search from '../reducers/search'

const reducers = combineReducers({
    List,
    Search
})

const store = createStore(reducers)

export default store