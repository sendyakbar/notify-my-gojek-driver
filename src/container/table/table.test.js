import React from 'react';
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer';

import Table from './table.jsx';
import ConnectedTabe from './table.jsx'
import { Provider } from 'react-redux';
import store from '../../store';

describe('<Table> component', () => {
    const wrapper = shallow(<Table />)
    test('render correctly', () => {
        expect(wrapper).toHaveLength(1)
    })
})

describe('<ConnectedTable /> redux-conneted component', () => {
    test('renders correctly', () => {
        const connectedTableWrapper = shallow(
            <Provider store={store}>
                <ConnectedTabe />
            </Provider>
        )
        expect(connectedTableWrapper).toHaveLength(1)
    })
    test('renders correctly and matched snapshot', () => {
        const tree = renderer.create(
            <Provider store={store}>
                <ConnectedTabe />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
})
