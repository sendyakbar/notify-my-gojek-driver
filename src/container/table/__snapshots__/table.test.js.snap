// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConnectedTable /> redux-conneted component renders correctly and matched snapshot 1`] = `
<div
  className="container"
>
  <table>
    <thead>
      <tr>
        <th>
          ID
        </th>
        <th>
          Name
        </th>
        <th>
          Phone
        </th>
        <th>
          Email
        </th>
        <th>
          Suspended
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td
          className="driverName"
        >
          John Doe
        </td>
        <td
          className="driverPhone"
        >
          +62888888888
        </td>
        <td
          className="driverEmail"
        >
          john@mail.com
        </td>
        <td>
          NO
        </td>
        <td>
          <div>
            <button
              className="button"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "grey",
                }
              }
            >
              Notify
            </button>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td
          className="driverName"
        >
          Jane Doe
        </td>
        <td
          className="driverPhone"
        >
          +62888888887
        </td>
        <td
          className="driverEmail"
        >
          jane@mail.com
        </td>
        <td>
          YES
        </td>
        <td />
      </tr>
      <tr>
        <td>
          3
        </td>
        <td
          className="driverName"
        >
          Oemar Bakrie
        </td>
        <td
          className="driverPhone"
        >
          +62888888886
        </td>
        <td
          className="driverEmail"
        >
          oemar@mail.com
        </td>
        <td>
          NO
        </td>
        <td>
          <div>
            <button
              className="button"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "grey",
                }
              }
            >
              Notify
            </button>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          4
        </td>
        <td
          className="driverName"
        >
          Johnny English
        </td>
        <td
          className="driverPhone"
        >
          +62888888885
        </td>
        <td
          className="driverEmail"
        >
          johnny@mail.com
        </td>
        <td>
          YES
        </td>
        <td />
      </tr>
      <tr>
        <td>
          5
        </td>
        <td
          className="driverName"
        >
          Ethan Hunt
        </td>
        <td
          className="driverPhone"
        >
          +62888888884
        </td>
        <td
          className="driverEmail"
        >
          ethan@mail.com
        </td>
        <td>
          NO
        </td>
        <td>
          <div>
            <button
              className="button"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "grey",
                }
              }
            >
              Notify
            </button>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          6
        </td>
        <td
          className="driverName"
        >
          Jaja Miharja
        </td>
        <td
          className="driverPhone"
        >
          +62888888883
        </td>
        <td
          className="driverEmail"
        >
          jaja@mail.com
        </td>
        <td>
          NO
        </td>
        <td>
          <div>
            <button
              className="button"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "grey",
                }
              }
            >
              Notify
            </button>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          7
        </td>
        <td
          className="driverName"
        >
          Rafael Benitez
        </td>
        <td
          className="driverPhone"
        >
          +62888888882
        </td>
        <td
          className="driverEmail"
        >
          rafael@mail.com
        </td>
        <td>
          NO
        </td>
        <td>
          <div>
            <button
              className="button"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "grey",
                }
              }
            >
              Notify
            </button>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          8
        </td>
        <td
          className="driverName"
        >
          Chris Martin
        </td>
        <td
          className="driverPhone"
        >
          +62888888881
        </td>
        <td
          className="driverEmail"
        >
          chris@mail.com
        </td>
        <td>
          NO
        </td>
        <td>
          <div>
            <button
              className="button"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "grey",
                }
              }
            >
              Notify
            </button>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          9
        </td>
        <td
          className="driverName"
        >
          Bon Jovi
        </td>
        <td
          className="driverPhone"
        >
          +62888888880
        </td>
        <td
          className="driverEmail"
        >
          bon@mail.com
        </td>
        <td>
          NO
        </td>
        <td>
          <div>
            <button
              className="button"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "grey",
                }
              }
            >
              Notify
            </button>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          10
        </td>
        <td
          className="driverName"
        >
          Iwan Fals
        </td>
        <td
          className="driverPhone"
        >
          +62888888879
        </td>
        <td
          className="driverEmail"
        >
          iwan@mail.com
        </td>
        <td>
          YES
        </td>
        <td />
      </tr>
    </tbody>
  </table>
  <div
    className="modal"
    id="myModal"
  >
    <div
      className="modal-content"
    >
      <p>
         
        
         
        
      </p>
      <div>
        <textarea
          onChange={[Function]}
          placeholder="Message"
          value=""
        />
      </div>
      <div
        className="buttonContainer"
      >
        <div>
          <button
            className="button"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#7BC415",
              }
            }
          >
            Send
          </button>
        </div>
        <div>
          <button
            className="button"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#DC6969",
              }
            }
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
